### PR TITLE
Clean up signal handling

### PR DESCRIPTION
**Pull request by Elarnon**
- We no longer use a post_save signal for updating scores when a Suggestion is
   changed; rather this has its place directly in the Suggestion.save method.
 - Signals are connected in the ready() method of Mangaki's new application
   configuration class, as recommended by Django's documentation:
   https://docs.djangoproject.com/en/dev/topics/signals/#connecting-receiver-functions
   Note that we use a receivers module instead of a signals module to allow for
   the creation and import of custom signals without registering handlers if
   the need ever arises.
 - Profile creation is no longer tied to login with django-allauth but rather
   to the actual User model creation. This helps handling corner cases such as
   accounts created through the `manage.py createsuperuser` management command
   actually having a profile.

### CODE REVIEW

File: mangaki/mangaki/apps.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from django.apps import AppConfig
+
+
+class MangakiConfig(AppConfig):
+    name = 'mangaki'
+    verbose_name = 'Mangaki'
+
+    def ready(self):
+        from . import receivers
```
Comment (RaitoBezarius) [inline mangaki/mangaki/apps.py]: Could you explain what does this line do in a comment? It is not that evident, I think.

File: mangaki/mangaki/views.py [hunk 2]
```diff
@@ -696,13 +694,6 @@ def add_pairing(request, artist_id, work_id):
     return HttpResponse()
 
 
-@receiver(user_signed_up)
-@receiver(social_account_added)
```
Comment (RaitoBezarius) [inline mangaki/mangaki/views.py]: Do we even support the ability to link new social accounts after being registered on Mangaki?

File: mangaki/mangaki/factories.py [hunk 3]
```diff
@@ -21,7 +21,6 @@ class Meta:
 
     username = factory.Faker('user_name')
     email = factory.LazyAttribute(lambda o: '{}@mangaki.fr'.format(o.username))
-    profile = factory.RelatedFactory(ProfileFactory, 'user')
```
Comment (RaitoBezarius) [inline mangaki/mangaki/factories.py]: Why it is unnecessary to have a `ProfileFactory` ?

Comment (Elarnon) [inline mangaki/mangaki/factories.py]: Thanks to the new signal, a new `Profile` will be created automatically for the user.

File: mangaki/mangaki/views.py (hunk 2 above)
Comment (Elarnon) [inline mangaki/mangaki/views.py]: I don't think so. In any case, the `Profile` has nothing to do with the existence of a social account.

File: mangaki/mangaki/factories.py (hunk 3 above)
Comment (RaitoBezarius) [inline mangaki/mangaki/factories.py]: Agreed, can we customize the data inside of the new profile? (in order to generate profiles in different "configuration")?

File: mangaki/mangaki/apps.py (hunk 1 above)
Comment (Elarnon) [inline mangaki/mangaki/apps.py]: It ensures that the `mangaki.receivers` module is loaded, which ensures that the signal receivers decorated with `@receiver` are connected when the app is ready.

File: mangaki/mangaki/factories.py (hunk 3 above)
Comment (Elarnon) [inline mangaki/mangaki/factories.py]: Actually this is the wrong way to go about it I believe, updating the patch to use `factory.django.mute_signals`.

Comment (RaitoBezarius) [inline mangaki/mangaki/factories.py]: Alright, thank you!

**Pull request merged**